src/actions/vstudio/vs2005_csproj.lua

```diff
@@ -314,7 +319,61 @@
 				_x(2,'<Reference Include="%s" />', name)
 			end
 		end)
+	end
+
+	function cs2005.nuGetReferences(prj)
+		if _ACTION >= "vs2010" and prj.nuget then
+			for i = 1, #prj.nuget do
+				local package = prj.nuget[i]
+				_x(2, '<Reference Include="%s">', vstudio.nuget2010.packageId(package))
+
+				local targetFramework = vstudio.nuget2010.packageFramework(prj.solution, package)
+
+				-- We need to write HintPaths for all supported framework
+				-- versions. The last HintPath will override any previous
+				-- HintPaths (if the condition is met that is).
+
+				local function writeHintPath(frameworkVersion)
+					local assembly = vstudio.path(
+						prj,
+						p.filename(
+							prj.solution,
+							string.format(
+								"packages\\%s\\lib\\%s\\%s.dll",
+								vstudio.nuget2010.packageName(package),
+								frameworkVersion,
+								vstudio.nuget2010.packageId(package)
+							)
+						)
+					)
+
+					_x(3, '<HintPath Condition="Exists(\'%s\')">%s</HintPath>', assembly, assembly)
+				end
+
+				for k, frameworkVersion in ipairs(vstudio.frameworkVersions) do
```

Comment:
It would be good to pull this logic out into its own function, both to make it more clear what it is doing and also so we can reuse it elsewhere if needed. Maybe something like `vstudio.identifyFrameworkVersion(prj)`?